packages/mobx/__tests__/v5/base/make-observable.ts

```diff
@@ -223,6 +224,39 @@ test("makeObservable supports autoBind", () => {
     expect(t.bound.call(undefined)).toBe(t)
 })
 
+test("Extending builtins is not support #2765", () => {
+    class ObservableMapLimitedSize extends ObservableMap {
+        limitSize = 0
+        constructor() {
+            super()
+            makeObservable(this, {
+                limitSize: observable
+            })
+        }
+    }
+
+    class ObservableSetLimitedSize extends ObservableSet {
+        limitSize = 0
+        constructor() {
+            super()
+            makeObservable(this, {
+                limitSize: observable
+            })
+        }
+    }
+
+    expect(() => {
+        new ObservableMapLimitedSize()
+    }).toThrowErrorMatchingInlineSnapshot(
+        `"[MobX] Cannot convert 'ObservableMap@10' into observable object: The target is already observable of different type. Extending builtins is not supported."`
```

Comment:
Please pass stable name: `makeObservable(_,_, { name: "TestObject" })` otherwise the tests are fragile - they break when you move them.

Reply:
OK, I noticed it actually. Thank you for fast reply. Haha